app/scripts/directives/service-definition-edit.js

```diff
@@ -80,11 +83,42 @@
         scope.transformToYamlDocumentFragment = function (editedJson) {
           var yamlFrag = {};
           angular.forEach(editedJson, function(svalue) {
-            yamlFrag[svalue.name] = svalue.value;
+            if (svalue.name === 'environment') {
+              yamlFrag[svalue.name] = parseEnvironmentVars(svalue);
+            } else {
+              yamlFrag[svalue.name] = svalue.value;
+            }
           });
           return yamlFrag;
         };
 
+        var parseEnvironmentVars = function (svalue) {
```

Comment:
This should be a more meaningful method name.  What's the reason for parsing the environment variables?